### PR TITLE
XDUser & Center[Director|Staff]Role Tests

### PR DESCRIPTION
**Pull request by ryanrath**
## Description
Adding component tests that exercise the functions in XDUser, CenterDirectorRole and CenterStaffRole that still use the `Role`, `UserRoles` and `UserRoleParameters` tables. Some of these new tests utilize documents that have been added to the `xdmod-test-artifacts` repo. 

## Motivation and Context
Expand the test coverage for one of our major classes: `XDUser`. 

CenterDirector and CenterStaff were included as they utilize the table `UserRoleParameters`.

## Tests performed
... all of them? 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [X] Additional Tests Added

## Checklist:

- [X] My code follows the code style of this project as found in the **CONTRIBUTING** document.
- [X] I have added tests to cover my changes.
- [X] All new and existing tests passed.

### CODE REVIEW

File: classes/User/Roles/CenterDirectorRole.php [hunk 1]
```diff
@@ -152,15 +152,6 @@ public function downgradeStaffMember($member)
 
         $pdo = DB::factory('database');
 
-        $is_primary_role = (
```
Comment (plessbd) [inline classes/User/Roles/CenterDirectorRole.php]: are these removals needed for the tests to work?  seems like this is not related to the tests

Comment (ryanrath) [reply classes/User/Roles/CenterDirectorRole.php]: Yup, if they are there then the `XDUserTest::testDowngradeStaffMemberSaveUser` test fails ( As there isn't a `primaryRole` from which to `getIdentifier()`  from ).
